<launch>

    <!-- Iris 2 -->
    <param name="IRIS_BODY_NAME" type="str" value="Iris2"/>
    <param name="LOAD_BODY_NAME" type="str" value="Load1"/>

    <param name="quadrotor_mass" type="double" value="1.442"/>
    <!-- no load -->
    <!-- <param name="total_mass" type="double" value="1.442"/> -->

        <!-- load lifting: cable length in meters -->
    <param name="cable_length" value="0.8"/> 
    <!-- load lifting: load mass in kg -->
    <param name="load_mass" value="0.145"/>
    <!-- with load -->
    <param name="total_mass" type="double" value="1.587"/>


    <!-- Iris 2 parameters -->
    <param name="proportional_gain_xy" type="double" value="4.0"/>
    <!-- <param name="proportional_gain_xy" type="double" value="6.0"/> -->
    <param name="derivative_gain_xy" type="double" value="6.0"/>
    <param name="integral_gain_xy" type="double" value="0.0"/>
    <param name="bound_integral_xy" type="double" value="0.0"/>

    <param name="natural_frequency_xy" type="double" value="2.0"/>
    <param name="damping_xy" type="double" value="1.5"/>
    <param name="position_saturation_xy" type="double" value="1.0"/>
    <param name="velocity_saturation_xy" type="double" value="1.0"/>


    <param name="proportional_gain_z" type="double" value="1.0"/>
    <param name="derivative_gain_z" type="double" value="1.414"/>
    <!-- <param name="integral_gain_z" type="double" value="0.4"/> -->
    <!-- <param name="bound_integral_z" type="double" value="0.0"/> -->
    <param name="integral_gain_z" type="double" value="0.0"/>
    <param name="bound_integral_z" type="double" value="0.5"/>

    <param name="natural_frequency_z" type="double" value="1.0"/>
    <param name="damping_z" type="double" value="0.707"/>
    <param name="position_saturation_z" type="double" value="0.5"/>
    <param name="velocity_saturation_z" type="double" value="0.5"/>    


    <param name="complete_neutral_reset" type="bool" value="false"/>
    <param name="throttle_neutral" type="double" value="1420"/>
    <param name="roll_neutral_value" type="double" value="1452"/>
    <param name="pitch_neutral_value" type="double" value="1465"/>
    <param name="iris_roll_gain" type="double" value="1.0"/>
    <param name="iris_pitch_gain" type="double" value="1.5"/>



    <!-- Controller Node -->
<!--     <node pkg="quad_control" name="plot_nodeasas" type="plot_experiment.py"  output="screen"/>
 -->
    <!-- Default Mission -->
    <param name="MissionDefault" type="str" value="IrisRealTrajectoryTracking"/>

    <!-- Default Yaw Controller -->
    <param name="YawControllerDefault" type="str" value="SimpleTrackingYawController"/>

    <!-- Default Trajectory -->
    <param name="TrajectoryDefault" type="str" value="StayAtRest"/>
    <param name="trajectry_offset_x" type="double" value="0.0"/>
    <param name="trajectry_offset_y" type="double" value="0.0"/>
    <param name="trajectry_offset_z" type="double" value="0.0"/>


    <!-- Default Yaw Controller -->
    <param name="ControllerDefault" type="str" value="SimplePIDController"/>
    <param name="DIControllerDefault" type="str" value="ComponentWise3DDIC"/>


    <!-- Default Vector Thrust Controller -->
    <param name="VThrustControllerDefault" type="str" value="VThrustQuadrupleController"/>

    <!-- Default Quadruple Integrator Controller -->
    <param name="QuadrupleIntegratorController" type="str" value="LinearQuadrupleIntController"/>

    <!-- Default Controller for Load transportation by single aerial vehicle -->
    <param name="SingleLoadTransportationControllerDefault" type="str" value="SingleLoadTransportController"/>

    <!-- Controller Node -->
    <node pkg="quad_control" name="controller_Iris2" type="cycle_quad_control_mission.py" output="screen"/>

    <!-- Controller parameters -->
    <param name="gravity_ctr" value="9.81"/>
    <param name="mass_quad_ctr" value="1.442"/>
    <param name="Throttle_neutral_ctr" value="1484.0"/>  
    <param name="ACRO_RP_P_ctr" value="4.5"/> 
    <param name="ktt" value="20.0"/> 
    <param name="ktt2" value="30.0"/>
    <!--  SOME USEFUL RULES: kp = wn*wn  AND kv = 2 xsi wn--> 
    <param name="kv" value="1.0"/> 
    <param name="kp" value="1.4"/> 
    <param name="sigma_p" value="0.5"/> 
    <param name="sigma_v" value="0.5"/> 

    <!-- <param name="quadrotor_mass" value="1.442"/> -->

    <!-- PARAMETERS THAT ARE COMMOM BETWEEN CONTROLLER AND SIMULATOR -->
    <!-- see WEB: ARDUCOPTER DIFFERENT MODES PARAMETERS -->
    <param name="MAX_ANGLE_DEG" value="45.0"/> 
    <param name="MAX_PSI_SPEED_Deg" value="200.0"/> 

    <!-- GUI -->
    <!-- <node name="rqt_gui" pkg="gui" type="tabbedGUI"/> -->
    <!-- <node name="rqt_gui" pkg="gui" type="tabPlugin"/> -->

    <!-- This runs rqt and we have to select plugin manually -->
    <!-- similar to rosrun rqt_gui rqt_gui -->
    <!-- <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui"/> -->
    <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="--standalone  tabbedGUI --args '' 'mocap'" output="screen"/>

    <node pkg="quad_control" name="service_time_sequencer" type="service_time_sequencer.py"  output="screen"/>

</launch>